### PR TITLE
Add shape and dtype for random uniform and dropout rewriter

### PR DESCRIPTION
**Pull request by csteegz**
Add shape and dtype for random uniform and dropout rewriter.

Fixes specific case of #304

### CODE REVIEW

File: tf2onnx/rewriter/random_uniform.py [hunk 1]
```diff
@@ -1,77 +1,79 @@
...
-        new_node = g.make_node("RandomUniformLike", inputs=[shape_node.input[0]], name=op_name,
-                               attr={"low": tmin, "high": tmax, "dtype": dtype})
-    else:
-        shape = g.get_shape(output.output[0])
-        new_node = g.make_node("RandomUniform", [], name=op_name,
-                               attr={"low": tmin, "high": tmax, "dtype": dtype, "shape": shape})
-    return new_node
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license.
+
```
Comment (pengwa) [inline tf2onnx/rewriter/random_uniform.py]: are you changing spaces to tab for the whole file?

File: tests/test_cond.py [hunk 2]
```diff
@@ -273,18 +273,16 @@ def test_dropout(self):
...
-        x = tf.random.uniform(x_val.shape, 0, x, seed=42)
+        x = tf.random.uniform([3], 0, 10, seed=42)
         y = y + 1
         res = tf.case([(tf.reduce_all(x < 1), lambda: x + y), (tf.reduce_all(y > 0), lambda: tf.square(y))],
                       default=lambda: x, name="test_case")
         _ = tf.identity(res, name="output")
 
-        feed_dict = {"input_1:0": x_val, "input_2:0": y_val}
-        input_names_with_port = ["input_1:0", "input_2:0"]
+        feed_dict = {"input_2:0": y_val}
```
Comment (pengwa) [inline tests/test_cond.py]: input_1:0 is should not be deleted, right?

File: tests/test_cond.py [hunk 3]
```diff
@@ -258,6 +258,35 @@ def case_graph():
         output_names_with_port = ["output:0"]
         self.run_test_case(feed_dict, input_names_with_port, output_names_with_port)
 
+    def test_dropout(self):
```
Comment (pengwa) [inline tests/test_cond.py]: pls put the common UT cases in test_backend.py.

**Comment by pengwa** [comment]
looked at the error for the first commit:

2019-02-13T22:33:02.7944464Z tf2onnx/rewriter/random_uniform.py:30: in rewrite_random_uniform
2019-02-13T22:33:02.7944548Z     tmax = input2.inputs[0].get_tensor_value()

tmax should be given a constant value, otherwise, the above logic cannot get the tensor. 
So in your UT case, the tmax should be a Constant instead of placeholder.

**Comment by pengwa** [comment]
this is converted by PR https://github.com/onnx/tensorflow-onnx/pull/318. feel free to have a try. if not work, let us know.
